# 세팅

In [ ]:
from google.colab import drive
drive.mount('/content/drive',force_remount=True)

Mounted at /content/drive


In [ ]:
import numpy as np
import pandas as pd
import glob
import pickle
%cd /content/drive/MyDrive/modeling/crawling_naver/

# 합치기

## csv 모두 합치기

이거 진행할 때에 /modeling/crawling_naver/publicdata_cafe_fornaver.csv 이거는 다른 곳으로 이동시키고 합쳐야 함!!!

In [ ]:
csv_list = sorted(glob.glob('*.csv'))
csv_list

['naver0-600.csv',
 'naver10000-10500.csv',
 'naver10500-13537.csv',
 'naver1200-2000.csv',
 'naver13538-14000.csv',
 'naver14000-14300.csv',
 'naver14300-14800.csv',
 'naver14800-15300.csv',
 'naver15300-15500.csv',
 'naver15500-16000.csv',
 'naver16000-16500.csv',
 'naver16500-17000.csv',
 'naver17000-17500.csv',
 'naver17500-18000.csv',
 'naver18000-18500.csv',
 'naver18500-18800.csv',
 'naver18800-19100.csv',
 'naver19100-.csv',
 'naver2000-2750.csv',
 'naver2750-3500.csv',
 'naver3500-4000.csv',
 'naver4000-4500.csv',
 'naver4500-5000.csv',
 'naver5000-5500.csv',
 'naver5500-6000.csv',
 'naver600-1200.csv',
 'naver6000-6500.csv',
 'naver6500-7000.csv',
 'naver7000-7400.csv',
 'naver7400-7500.csv',
 'naver7500-7900.csv',
 'naver7900-8000.csv',
 'naver8000-8500.csv',
 'naver8500-9000.csv',
 'naver9000-9500.csv',
 'naver9500-10000.csv']

In [ ]:
csv_list = sorted(glob.glob('*.csv'))
df = pd.DataFrame(columns = ['cafe', 'search_name', 'address','road_address','near_station','station_distance','total_review',
                  '"커피가 맛있어요"','"친절해요"','"음료가 맛있어요"', '"디저트가 맛있어요"',
                  '"매장이 청결해요"', '"대화하기 좋아요"', '"집중하기 좋아요"','"인테리어가 멋져요"',
                  '"좌석이 편해요"','"화장실이 깨끗해요"','"특별한 메뉴가 있어요"','"가성비가 좋아요"',
                  '"뷰가 좋아요"', '"주차하기 편해요"', '"사진이 잘 나와요"'])
for csv in csv_list:
  new = pd.read_csv(csv)
  df = pd.concat([df, new], ignore_index= True)
  print(len(new), '개의 행 생성! 총', len(df), '개의 행이 존재합니다.')



212 개의 행 생성! 총 212 개의 행이 존재합니다.
183 개의 행 생성! 총 395 개의 행이 존재합니다.
1184 개의 행 생성! 총 1579 개의 행이 존재합니다.
371 개의 행 생성! 총 1950 개의 행이 존재합니다.
118 개의 행 생성! 총 2068 개의 행이 존재합니다.
110 개의 행 생성! 총 2178 개의 행이 존재합니다.
223 개의 행 생성! 총 2401 개의 행이 존재합니다.
244 개의 행 생성! 총 2645 개의 행이 존재합니다.
113 개의 행 생성! 총 2758 개의 행이 존재합니다.
266 개의 행 생성! 총 3024 개의 행이 존재합니다.
260 개의 행 생성! 총 3284 개의 행이 존재합니다.
235 개의 행 생성! 총 3519 개의 행이 존재합니다.
267 개의 행 생성! 총 3786 개의 행이 존재합니다.
294 개의 행 생성! 총 4080 개의 행이 존재합니다.
291 개의 행 생성! 총 4371 개의 행이 존재합니다.
178 개의 행 생성! 총 4549 개의 행이 존재합니다.
182 개의 행 생성! 총 4731 개의 행이 존재합니다.
328 개의 행 생성! 총 5059 개의 행이 존재합니다.
371 개의 행 생성! 총 5430 개의 행이 존재합니다.
307 개의 행 생성! 총 5737 개의 행이 존재합니다.
184 개의 행 생성! 총 5921 개의 행이 존재합니다.
206 개의 행 생성! 총 6127 개의 행이 존재합니다.
206 개의 행 생성! 총 6333 개의 행이 존재합니다.
199 개의 행 생성! 총 6532 개의 행이 존재합니다.
226 개의 행 생성! 총 6758 개의 행이 존재합니다.
255 개의 행 생성! 총 7013 개의 행이 존재합니다.
210 개의 행 생성! 총 7223 개의 행이 존재합니다.
183 개의 행 생성! 총 7406 개의 행이 존재합니다.
156 개의 행 생성! 총 7562 개의 행이 존재합니다.
51 개의 행 생성! 총 7613 개의 행이 존재합니다.
158 개의 행 생성!

In [ ]:
df.head()

cafe search_name address                road_address  \
0                  버블베어        버블베어     방화동  서울 강서구 금낭화로 167 방화8단지개화아파트   
1  커피빈코리아낙성대역점 코리아낙성대역점   커피빈 낙성대역점     봉천동      서울 관악구 남부순환로 1909 1,2층   
2                 노란고무줄  노란꼬무줄 선릉본점     대치동   서울 강남구 선릉로 424 1층5호 노란꼬무줄   
3                   커피랩         커피랩     서교동           서울 마포구 와우산로29길 14   
4         커핀그루나루신천점 신천점  커핀그루나루 신천점     잠실동            서울 송파구 올림픽로12길 8   

   near_station station_distance total_review  "커피가 맛있어요" "친절해요"  "음료가 맛있어요"  \
0          5방화역             427m           18         7.0      5         5.0   
1         2낙성대역             119m          284         7.0     56        74.0   
2      2수인분당선릉역             157m          134        40.0     14        61.0   
3  2공항경의중앙홍대입구역             324m          100        76.0     23        13.0   
4        2잠실새내역             228m          140        64.0     14        31.0   

   ...  "대화하기 좋아요"  "집중하기 좋아요"  "인테리어가 멋져요"  "좌석이 편해요"  "화장실이 깨끗해요"  \
0  ...         NaN         NaN          NaN        1.0          2.0   
1  ...        21.0        48.0          8.0       33.0          2.0   
2  ...         NaN         NaN          1.0        NaN          NaN   
3  ...        17.0        10.0         19.0        2.0          6.0   
4  ...        29.0         7.0          2.0       15.0          5.0   

   "특별한 메뉴가 있어요"  "가성비가 좋아요"  "뷰가 좋아요"  "주차하기 편해요"  "사진이 잘 나와요"  
0            2.0         NaN       NaN         1.0          NaN  
1            2.0         8.0      14.0         NaN          1.0  
2           21.0        33.0       NaN         1.0          NaN  
3           10.0         4.0       5.0         NaN          4.0  
4            5.0         6.0       3.0         NaN          1.0  

[5 rows x 22 columns]

In [ ]:
# 지금 변수명들이 직관적이지 않아서 약간 수정을 해보려고 합니다.
df.rename(columns={'cafe':'name'},inplace=True)


In [ ]:
df.to_csv('naver_crawled_total.csv',index=False)

# csv 전처리

## 카페가 아닌 경우 제거하기

In [ ]:
%cd /content/drive/MyDrive/modeling/crawling_naver/
df = pd.read_csv('naver_crawled_total.csv')
df.info()

/content/drive/.shortcut-targets-by-id/1wuSyaB4gVB_OYC11vUBFSUjLBBeITS80/modeling/crawling_naver
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8496 entries, 0 to 8495
Data columns (total 23 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              8496 non-null   object 
 1   search_name       8496 non-null   object 
 2   address           8496 non-null   object 
 3   road_address      7619 non-null   object 
 4   near_station      7619 non-null   object 
 5   station_distance  7619 non-null   object 
 6   total_review      8496 non-null   int64  
 7   "커피가 맛있어요"        7798 non-null   float64
 8   "친절해요"            8468 non-null   float64
 9   "음료가 맛있어요"        8024 non-null   float64
 10  "디저트가 맛있어요"       7707 non-null   float64
 11  "매장이 청결해요"        8187 non-null   float64
 12  "대화하기 좋아요"        7418 non-null   float64
 13  "집중하기 좋아요"        6244 non-null   float64
 14  "인테리어가 멋져요"       7659 non-null   float

### 키워드가 전부 NaN인 경우(아예 다른 분야)

In [ ]:
# 리뷰 칸은 총 10명이상이 참여해야 열림
# => 그런데 리뷰칸이 전부 NaN이라는 것은 그 매장이 카페가 아니라는 것을 의미함.
# 행 제거 필요

# 결과를 보니 전부 학원 책방이어서 제거해야함

df[(df['"커피가 맛있어요"'].isnull()) & (df['"친절해요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"매장이 청결해요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"인테리어가 멋져요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"화장실이 깨끗해요"'].isnull()) & (df['"특별한 메뉴가 있어요"'].isnull()) & (df['"가성비가 좋아요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"주차하기 편해요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())]

name       search_name address  \
231   카페바리스타아카데미        카페바리스타아카데미     역삼동   
2476       이나영책방             이나영책방     신림동   
2538     하우 신림역점   하우스터디독서실 신림난곡센터     신림동   
3243     로허들커피교실          로허들 커피교실     신정동   
6531       이루리북스             이루리북스     도화동   
6708    진부책방스튜디오          진부책방스튜디오     서교동   
6747        한평책빵              한평책빵     녹번동   
6985   카페다커피 역촌점           카페원커피학원     역촌동   
7208      위트앤시니컬     시집서점 위트 앤 시니컬     혜화동   
8038       지금의세상            지금의 세상     사당동   
8121    잇츠리얼타임카페  잇츠리얼타임 독서실 스터디카페    노량진동   
8338  공차 영풍문고종로점         영풍문고 종로본점     서린동   

                             road_address near_station station_distance  \
231                     서울 강남구 테헤란로10길 43      2신분당강남역             527m   
2476                   서울 관악구 신사로 66-1 3층        2신대방역             360m   
2538                                  NaN          NaN              NaN   
3243                 서울 양천구 신정중앙로 63 207호         5신정역             335m   
6531         서울 마포구 독막로 320 태영데시앙루브 B105호  56공항경의중앙공덕역             353m   
6708                   서울 마포구 잔다리로 112 2층         6망원역             412m   
6747  서울 은평구 통일로 684 서울혁신파크 25동 101호 한평책빵        36불광역             123m   
6985                       서울 은평구 가좌로 249         6새절역             214m   
7208          서울 종로구 창경궁로 271-1 동양서림 내 2층         4혜화역             257m   
8038                  서울 동작구 동작대로3길 41 1층        24사당역             235m   
8121           서울 동작구 노량진로 140 메가스터디타워 5층       19노량진역              79m   
8338                       서울 종로구 청계천로 41         1종각역              15m   

      total_review  "커피가 맛있어요"  "친절해요"  "음료가 맛있어요"  ...  "집중하기 좋아요"  \
231             16         NaN     NaN         NaN  ...         NaN   
2476            10         NaN     NaN         NaN  ...         NaN   
2538            21         NaN     NaN         NaN  ...         NaN   
3243            75         NaN     NaN         NaN  ...         NaN   
6531            23         NaN     NaN         NaN  ...         NaN   
6708            35         NaN     NaN         NaN  ...         NaN   
6747           123         NaN     NaN         NaN  ...         NaN   
6985            19         NaN     NaN         NaN  ...         NaN   
7208            13         NaN     NaN         NaN  ...         NaN   
8038            22         NaN     NaN         NaN  ...         NaN   
8121           410         NaN     NaN         NaN  ...         NaN   
8338          2814         NaN     NaN         NaN  ...         NaN   

      "인테리어가 멋져요"  "좌석이 편해요"  "화장실이 깨끗해요"  "특별한 메뉴가 있어요"  "가성비가 좋아요"  \
231           NaN        NaN          NaN            NaN         NaN   
2476          NaN        NaN          NaN            NaN         NaN   
2538          NaN        NaN          NaN            NaN         NaN   
3243          NaN        NaN          NaN            NaN         NaN   
6531          NaN        NaN          NaN            NaN         NaN   
6708          NaN        NaN          NaN            NaN         NaN   
6747          NaN        NaN          NaN            NaN         NaN   
6985          NaN        NaN          NaN            NaN         NaN   
7208          NaN        NaN          NaN            NaN         NaN   
8038          NaN        NaN          NaN            NaN         NaN   
8121          NaN        NaN          NaN            NaN         NaN   
8338          NaN        NaN          NaN            NaN         NaN   

      "뷰가 좋아요"  "주차하기 편해요"  "사진이 잘 나와요"   same  
231        NaN         NaN          NaN   True  
2476       NaN         NaN          NaN   True  
2538       NaN         NaN          NaN  False  
3243       NaN         NaN          NaN  False  
6531       NaN         NaN          NaN   True  
6708       NaN         NaN          NaN   True  
6747       NaN         NaN          NaN   True  
6985       NaN         NaN          NaN  False  
7208       NaN         NaN          NaN  False  
8038       NaN         NaN          NaN  False  
8121       NaN         NaN          N

In [ ]:
all_nan_index = df[(df['"커피가 맛있어요"'].isnull()) & (df['"친절해요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"매장이 청결해요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"인테리어가 멋져요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"화장실이 깨끗해요"'].isnull()) & (df['"특별한 메뉴가 있어요"'].isnull()) & (df['"가성비가 좋아요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"주차하기 편해요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())].index

In [ ]:
# 제거 완료
df = df.drop(all_nan_index)

### 키워드를 하나씩 조건에서 제거해보면서 제거할 행 찾기

In [ ]:
# 근데 예외경우가 또 있을 수도 있다! => 방탈출 카페나 다른 곳에 대해서도 친절해요 주차하기 편해요가 있을 수도 있어서 그 경우는 또는 통해서 계속 소거해 나가야 겠다.

# 먼저 친절해요는 조건에서 제해볼까?

# 이경우 꽃집(붉은산다화, 애쉴리즈, 아르넬), 방탈출카페(비트포비아, ), 
# 공방(반지카페), 장난감(덕카페, 루나블랑,). 헬스장(안녕 휘트니스) 이 나옴
# 전부 제거해도 됨

df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"매장이 청결해요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"인테리어가 멋져요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"화장실이 깨끗해요"'].isnull()) & (df['"특별한 메뉴가 있어요"'].isnull()) & (df['"가성비가 좋아요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"주차하기 편해요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())]


name       search_name address  \
118             붉은산다화             붉은산다화     성내동   
394              애쉴리즈              애쉴리즈     청담동   
676         비트포비아 강남점        비트포비아 강남던전     역삼동   
899           반지카페8그램    반지만들기 반지공방 8그램     역삼동   
952               덕카페               덕카페    상월곡동   
1675        아르넬ARENEL               아르넬     신사동   
1802             루나블랑              루나블랑     신당동   
1902      비트포비아 던전홍대점        비트포비아 홍대던전     서교동   
1935       키이스케이프 홍대점        키이스케이프 홍대점     서교동   
2149             샤론화원              샤론화원     신정동   
2356       넥스트에디션 잠실점  방탈출카페 넥스트에디션 잠실점     잠실동   
2485              조르막               조르막     반포동   
2605             아일랑트   아일랑트 캔들&카페 홍대본점     서교동   
3154             티케이프              티케이프     창천동   
5071  비올즈플라워카페 비올즈플라워       비올즈플라워카페2호점     봉천동   
5379   도어이스케이프 레드신논현점   도어이스케이프 레드 신논현점     서초동   
5838         샌프란시스코커피         샌프란시스코플라워     신당동   
6342             개미지옥              개미지옥     논현동   
7180            라플레르수             라플레르수     부암동   
7843            마인드리움             마인드리움     마곡동   
7991          안녕,도화동.        안녕휘트니스 마포점     도화동   
8191     태무든TAEMOODEN               태무든     홍은동   

                                 road_address near_station station_distance  \
118                        서울 강동구 성내로3가길 7 1층       8강동구청역             176m   
394             서울 강남구 학동로77길 5 청담그랑빌102 애쉴리즈         7청담역             123m   
676      서울 강남구 강남대로84길 33 대우디오빌플러스 지하1층 111호      2신분당강남역             259m   
899                  서울 강남구 봉은사로2길 37 혜인빌딩 5층     9신분당신논현역             208m   
952                       서울 성북구 화랑로25길 19 2층        6상월곡역             316m   
1675                   서울 강남구 압구정로4길 14-10 1층      3신분당신사역             611m   
1802                      서울 중구 동호로17길 277 2층       6버티고개역             189m   
1902                     서울 마포구 독막로3길 30 지하2층        26합정역             296m   
1935             서울 마포구 어울마당로 44-1 보리수빌딩 지하1층         6상수역             314m   
2149  서울 양천구 목동동로8길 29 한양프라자 1층 Sharon Flower       2양천구청역             803m   
2356                     서울 송파구 올림픽로 118 지하1층       2잠실새내역              81m   
2485                     서울 서초구 반포대로39길 42 1층    379고속터미널역             675m   
2605                  서울 마포구 와우산로21길 19-19 1층         6상수역             480m   
3154                   서울 서대문구 연세로12길 8 1층 2층         2신촌역             383m   
5071            서울 관악구 남부순환로 1870 비올즈플라워카페 1층      2서울대입구역             414m   
5379                     서울 서초구 사평대로 360 지하1층     9신분당신논현역             135m   
5838                        서울 중구 다산로42길 43-4        26신당역             140m   
6342                   서울 강남구 논현로149길 71 104호      3신분당신사역             231m   
7180                                      NaN          NaN              NaN   
7843        서울 강서구 공항대로51길 17-7 1층 102호 마인드리움         9증미역             586m   
7991                     서울 마포구 마포대로 49 지하 1층         5마포역             118m   
8191                    서울 서대문구 세검정로1길 105 1층         3홍제역             895m   

      total_review  "커피가 맛있어요"  "친절해요"  "음료가 맛있어요"  ...  "집중하기 좋아요"  \
118             26         NaN    20.0         NaN  ...         NaN   
394             15         NaN     9.0         NaN  ...         NaN   
676             21         NaN     6.0         NaN  ...         NaN   
899            167         NaN     9.0         NaN  ...         NaN   
952             11         NaN     4.0         NaN  ...         NaN   
1675            13         NaN     8.0         NaN  ...         NaN   
1802            23         NaN    14.0         NaN  ...         NaN   
1902            14         NaN     4.0         NaN  ...         NaN   
1935            37         NaN    17.0         NaN  ...         NaN   
2149            27         NaN     8.0         NaN  ...         NaN   
2356            22         NaN     9.0         NaN  ...         NaN   
2485            27         NaN    10.0         NaN  ...         NaN   
2605            35         NaN    24.0         NaN  ...         NaN   
3154            13         NaN 

In [ ]:
except_kind = df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"매장이 청결해요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"인테리어가 멋져요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"화장실이 깨끗해요"'].isnull()) & (df['"특별한 메뉴가 있어요"'].isnull()) & (df['"가성비가 좋아요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"주차하기 편해요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())].index


In [ ]:
df = df.drop(except_kind)

In [ ]:
# 이번엔 주차하기 편해요랑 친절해요를 제거해보자

df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"매장이 청결해요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"인테리어가 멋져요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"화장실이 깨끗해요"'].isnull()) & (df['"특별한 메뉴가 있어요"'].isnull()) & (df['"가성비가 좋아요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())]

# 꽃집인데 이름이 커피, 한송이네 => 제거
# 공방 : 쓰임
# 꽃집 : 꽃카페JE, 오뉴월, 쉔, 오이나플라워앤커피바, 블루미시티, 르자당도산점, 소올소올, 어반그로브, 보나르떼, 블루밍테라스
# 학원: 프롱
# 사진, 스튜디오: 사진찍는카페찰리
# 방탈출: 던전 101

# 리퍼블릭은 제대로 못 불러옴 (카페 맞음)

name      search_name address  \
78        커피,한송이           커피 한송이     등촌동   
89            쓰임               쓰임     방학동   
442        꽃카페JE               쥬브     잠원동   
1531         오뉴월        플라워카페 오뉴월     마곡동   
2285    사진찍는카페찰리          찰리카페사진관     반포동   
2352       던전101      비트포비아 던전101     서교동   
3466           쉔             쉔플라워     천호동   
3553  오이나플라워앤커피바    오이나 플라워 앤 커피바     가락동   
5343       블루미시티            블루미시티     논현동   
5476          프롱  프롱센트럴바리스타제과제빵학원     흥인동   
6028       메리플레르            메리플레르     상계동   
6257        리퍼블릭             리퍼블릭     신사동   
6382     르자당 도산점          르자당 도산점     신사동   
6508        소올소올          소올소올플라워     고척동   
7134       어반그로브            어반그로브     공평동   
7136        보나르떼             보나르떼     신사동   
7235      블루밍테라스           블루밍테라스   하왕십리동   
7584   무토스아트그라운드       무토스 아트그라운드     석촌동   
7765   스노우폭스플라워스    스노우폭스플라워 선릉역점     삼성동   
8181     월계골프연습장       스파크월계골프연습장     월계동   
8262       블루블루엣            블루블루엣    여의도동   
8401     반지캠퍼스카페          반지캠퍼스카페     서교동   

                                road_address  near_station station_distance  \
78                  서울 강서구 공항대로45길 44 청림오피스텔          9가양역             541m   
89    서울 도봉구 도봉로143길 32 leather work-shop 쓰임          1방학역             806m   
442        경기 안양시 동안구 시민대로 230 아크로타워 c동 223호          4범계역             488m   
1531        서울 강서구 마곡중앙로 59-17 류마타워2 1층 116호          5마곡역             120m   
2285                        서울 서초구 동광로49길 88     379고속터미널역             861m   
2352                       서울 마포구 와우산로 74 2층          6상수역             356m   
3466                       서울 강동구 천중로 179 1층          5길동역             456m   
3553             서울 송파구 송이로26길 16-60 1층 101호        3경찰병원역             320m   
5343                       서울 강남구 논현로157길 28         3압구정역             597m   
5476                         서울 중구 마장로 66 2층         26신당역             188m   
6028                      서울 노원구 동일로 1661 1층         7수락산역              23m   
6257                                     NaN           NaN              NaN   
6382                  서울 강남구 도산대로45길 10-3 3층   수인분당압구정로데오역             673m   
6508                                     NaN           NaN              NaN   
7134            서울 종로구 우정국로 26 센트로폴리스빌딩 지하2층          1종각역              69m   
7136                    서울 강남구 논현로161길 41 1층         3압구정역             562m   
7235          서울 성동구 왕십리로 374 센트라스 130동 지하1층        2상왕십리역              41m   
7584             서울 송파구 송파대로37길 84 3층 아트그라운드        9석촌고분역             863m   
7765                      서울 강남구 테헤란로 403 1층      2수인분당선릉역              15m   
8181          서울 노원구 초안산로2길 81-38 스파크월계골프연습장          1월계역             388m   
8262     서울 영등포구 여의대로 56 한화손해보험빌딩 지하1층 블루블루엣        59여의도역             266m   
8401                      서울 마포구 어울마당로 112-6  2공항경의중앙홍대입구역             191m   

      total_review  "커피가 맛있어요"  "친절해요"  "음료가 맛있어요"  ...  "집중하기 좋아요"  \
78              52         NaN    24.0         NaN  ...         NaN   
89              20         NaN    14.0         NaN  ...         NaN   
442             28         NaN    18.0         NaN  ...         NaN   
1531            31         NaN    23.0         NaN  ...         NaN   
2285            95         NaN    76.0         NaN  ...         NaN   
2352            33         NaN    14.0         NaN  ...         NaN   
3466            55         NaN    34.0         NaN  ...         NaN   
3553           115         NaN    91.0         NaN  ...         NaN   
5343            81         NaN    57.0         NaN  ...         NaN   
5476            14         NaN     NaN         NaN  ...         NaN   
6028            30         NaN     9.0         NaN  ...         NaN   
6257           116         NaN    45.0         NaN  ...         NaN   
6382            55         NaN    31.0         NaN  ...         NaN   
6508            27         NaN    21.0         NaN  ...         NaN   
7134            17         NaN    13.0         NaN  ...         NaN   
7136            31    

In [ ]:
except_k_p = df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"매장이 청결해요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"인테리어가 멋져요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"화장실이 깨끗해요"'].isnull()) & (df['"특별한 메뉴가 있어요"'].isnull()) & (df['"가성비가 좋아요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())].index

df.drop(index=except_k_p, inplace=True)

In [ ]:
# 음료와 디저트 관련된 키워드 모든게 다 nan값이면 카페가 아닐수도 있다고 봐보자

# 보드게임카페: 하이힐, 더조커
# 머리염색: 더오색
# 음식점: 평통보쌈, 텃골(김밥), 동심
# 만화카페: 벌툰, 알타미라
# 편의점: 이마트24
# 스튜디오: 영상다방황금단추
# 방탈출: 비트포비아, 비밀의 화원, 마스터키, 포인트나인, 비밀의숲, 세븐클루스, 시크릿코드
# 인테리어 소품: 레이지 선데이
# 주류: 비틀비틀
# 메이크업: 
# 잘못 가져온 경우도 많이 걸러짐
df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"특별한 메뉴가 있어요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())][0:60]

name         search_name address  \
82             통툰만화까페         통툰만화까페 발산역점    내발산동   
105            놀숲 건대점              놀숲 건대점     자양동   
107         놀숲 플러스홍대점           놀숲플러스 홍대점     서교동   
233             덤스 본점        범스불고기 대구혁신본점     등촌동   
257   카툰공감성신여대점 성신여대점          카툰공감 성신여대점   동선동1가   
356               하이힐                 하이힐    이태원동   
359               더오색                 더오색     논현동   
443            개성평통보쌈            평통보쌈 공단점   금호동4가   
589                텃골                  텃골     연희동   
596            벌툰 신촌점              벌툰 신촌점     창천동   
669    이마트24 R가산하이엔드점      이마트24 R가산하이엔드점     가산동   
694          영상다방황금단추           영상다방 황금단추     합정동   
767       이마트 24종로당주점         이마트24 종로당주점     당주동   
778             포비아명동            비트포비아 명동    명동2가   
892            레이지선데이              레이지선데이   성수동1가   
895              알타미라           딩굴딩굴 알타미라     서교동   
1024       놀숲 신정네거리역점          놀숲 신정네거리역점     신정동   
1149        비밀의화원리버타운          비밀의화원 리버타운     역삼동   
1153       마스터키 홍대상수점          마스터키 홍대상수점     상수동   
1154            포인트나인               포인트나인     역삼동   
1171             비밀의숲               비밀의 숲     서교동   
1199        엑스케이프 홍대점           엑스케이프 홍대점     서교동   
1210      세븐클루스 강남2호점        세븐클루스 강남 2호점     역삼동   
1229        시크릿코드 홍대점         시크릿코드 홍대직영점     서교동   
1303       셜록홈즈 강남2호점          셜록홈즈 강남2호점     서초동   
1371        엑스케이프 건대점           엑스케이프 건대점     화양동   
1378              더조커      보드게임카페 더조커 홍대점     서교동   
1456        위드미 메리어트점         이마트24 메리어트점     반포동   
1596         리틀비틀 아현점                비틀비틀     아현동   
1606         놀숲 마곡발산점            놀숲 마곡발산점     마곡동   
1610     코드이스케이프 가로수길        코드이스케이프 가로수길     신사동   
1625              엘크레                 엘크레     신사동   
1698           여왕의브런치              여왕의브런치     역삼동   
1856       셜록홈즈 잠실2호점          셜록홈즈 잠실2호점     잠실동   
1889         제로월드 강남점            제로월드 강남점     서초동   
1901        비밀의화원 홍대점          비밀의화원 미드나잇     서교동   
1903             카페단색                반색식당     합정동   
1932   엑스이스케이프 홍대놀이터점      엑스이스케이프 홍대놀이터점     서교동   
2030      홍베이팡 잠실나루역점         홍베이팡 잠실나루역점     신천동   
2126               동심              한정식 동심   남산동2가   
2254      포퀸스보드카페 등촌점         포퀸스보드카페 등촌점     등촌동   
2294         마스터키 강남점         마스터키프라임 강남점     역삼동   
2296      세븐클루스 강남1호점         세븐클루스 강남1호점     역삼동   
2382      마스터키프라임 건대점        마스터키 프라임 건대점     자양동   
2455               담다                  담다     마곡동   
2534             유노이아                유노이아   양평동2가   
2588        주아킹위례 중앙점           주아킹 위례중앙점     장지동   
2775      GS25 마곡9단지점         GS25 마곡9단지점     마곡동   
2837          과일연구소당도            과일연구소 당도     풍납동   
3036           카페 애자일  크라임씬카페 퍼즐팩토리 홍대2호점     정릉동   
3298             엣지커피                엣지커피    남가좌동   
3417             허그댕댕                허그댕댕     창신동   
3458             화양연화                화양연화     화양동   
3616              삼키로                 삼키로     중계동   
3807             개성있냥                개성있냥     가락동   
5115        놀숲플러스 홍대점           놀숲플러스 홍대점     서교동   
5143         덤앤더머 홍대점      덤앤더머 방탈출카페 홍대점     서교동   
5169     위드미청계천점 청계천점          이마트24 청계천점     서린동   
5238  브레이크아웃이스케이프 홍대점         브레이크아웃이스케이프     서교동   
5294               쿼즈             쿼즈 보드카페      목동   

                                      road_address  near_station  \
82                                서울 강서구 강서로52길 16          5발산역   
105                       서울 광진구 아차산로 224 3층 놀숲건대점       27건대입구역   
107                            서울 마포구 양화로16길 29 B1  2공항경의중앙홍대입구역   
233                                            NaN           NaN   
257                         서울 성북구 동소문로20가길 4 지하1층  4우이신설성신여대입구역   
356                               서울 용산구 보광로59길 36         6이태원역   
359                       서울 강남구 봉은사로29길 21 강남성우빌딩          9언주역   
443                                            NaN           NaN   
589                                            NaN           NaN   
596                             서울 서대문구 연세로5가길 8-3          2신촌역   
669                      서울 금천구 벚꽃로

In [ ]:
# 귀금속 : 그루브앤골드, 트롤비즈코리아
# 편의점: gs25
# 음식점: 청년다방, 샐러디, 쩡이네(건어물)


df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"특별한 메뉴가 있어요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())][60:]

name   search_name address                    road_address  \
5312           이스케이프샵       이스케이프 샾     신사동          서울 강남구 논현로175길 93 지하1층   
5475       머더파커 건대2호점    머더파커 건대2호점     화양동           서울 광진구 동일로22길 68 지하1층   
5481           그루브앤골드    그루브앤골드 망원점     망원동              서울 마포구 망원로8길 67 1층   
5565       카페데코믹스 홍대점    카페데코믹스 홍대점     서교동              서울 마포구 어울마당로 96 3층   
5702       GS25 명륜성대점    GS25 명륜성대점    명륜1가                    서울 종로구 명륜길 2   
5858           만화카페통툰       통툰 영등포점  영등포동3가     서울 영등포구 영중로6길 12 포레스트힐시티 B1   
5993       놀숲 충무로동국대점    놀숲 충무로동국대점   충무로5가                서울 중구 서애로 6-1 3층   
6066       청년다방 오목교역점    청년다방 오목교역점     신정동       서울 양천구 목동동로12길 22 프리우스아파트   
6314      놀숲강남역점 강남역점       놀숲 강남역점     역삼동    서울 강남구 강남대로98길 17 1층 놀숲 강남역점   
6320        샐러디 양천구청점     샐러디 양천구청점     신정동     서울 양천구 목동서로 349 센트럴프라자 127호   
6589  GS25연대서문점 연대서문점    GS25 연대서문점     연희동                             NaN   
6622       통툰 서울대입구역점    통툰 서울대입구역점     봉천동  서울 관악구 관악로16길 13 5층 통툰 서울대입구역점   
6726              스모모           스모모     화양동           경기 남양주시 오남읍 진건오남로 880   
6793      놀숲둔촌역점 둔촌역점       놀숲 둔촌역점     성내동             서울 강동구 양재대로 1319 3층   
6814    놀숲압구정역점 압구정역점      놀숲 압구정역점     신사동       서울 강남구 압구정로28길 22-6 지하 1층   
6989      놀숲이수역점 이수역점       놀숲 이수역점     사당동     서울 동작구 동작대로27길 5 지안플라자 지하1층   
7003         셜록홈즈 홍대점      셜록홈즈 홍대점     서교동               서울 마포구 와우산로27길 35   
7137              쩡이네           쩡이네     오장동  서울 중구 을지로32길 39 (오장동) 중부시장 쩡이네   
7414          트롤비즈코리아  트롤비즈코리아 서울숲점   성수동1가              서울 성동구 서울숲4길 24 1층   
7440        위드미 홍대서교점   이마트24 홍대서교점     서교동               서울 마포구 와우산로29길 48   
7681      놀숲사당역점 사당역점       놀숲 사당역점     남현동              서울 관악구 과천대로 941 6층   
8057    놀숲숙대입구점 숙대입구점      놀숲 숙대입구점   청파동2가           서울 용산구 청파로47길 52 지하1층   
8082           벌툰 당산점        벌툰 당산점   당산동6가       서울 영등포구 양평로 18 2층 벌툰 만화카페   
8092      놀숲신림역점 신림역점       놀숲 신림역점     신림동                  서울 관악구 신림로 321   
8369       놀숲 강남교보타워점    놀숲 강남교보타워점     서초동    서울 서초구 서초대로77길 58 트라이앤드빌딩 B1   
8380               해빛    해빛 한강잠원1호점     안국동                             NaN   
8454        안녕강아지와고양이     안녕강아지와고양이     오금동                   서울 송파구 마천로 99   

       near_station station_distance  total_review  "커피가 맛있어요"  "친절해요"  \
5312          3압구정역             733m           220         NaN     4.0   
5475        27건대입구역             151m            24         NaN     7.0   
5481           6망원역             439m            11         NaN     4.0   
5565   2공항경의중앙홍대입구역             376m            51         NaN    14.0   
5702           4혜화역             838m           126         NaN    74.0   
5858          1영등포역             312m            96         NaN    16.0   
5993         34충무로역             300m            26         NaN    11.0   
6066          5오목교역             259m            43         NaN    10.0   
6314        2신분당강남역             333m            63         NaN    25.0   
6320         2양천구청역             819m            15         NaN     2.0   
6589            NaN              NaN            44         NaN     7.0   
6622        2서울대입구역              95m            20         NaN     8.0   
6726           4오남역             556m            19         NaN    15.0   
6793          5둔촌동역             404m            48         NaN    20.0   
6814          3압구정역             160m            28         NaN     9.0   
6989          47이수역             106m            20         NaN     5.0   
7003   2공항경의중앙홍대입구역             363m            12         NaN     3.0   
7137  245동대문역사문화공원역             344m            39         NaN    11.0   
7414           2뚝섬역             361m            30         NaN    18.0   
7440   2공항경의중앙홍대입구역             190m            44         NaN    17.0   
7681          24사당역              49m            35         NaN     6.0   
8057         4숙대입구역             473m            50         NaN    10.0   
8082          29당산역              21m            27         NaN     6.0   
8092         2신림신림역         

In [ ]:
# 위의 경우 전부 제거해도 된다.
index_3 = df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) & (df['"좌석이 편해요"'].isnull()) & 
   (df['"특별한 메뉴가 있어요"'].isnull()) & 
   (df['"뷰가 좋아요"'].isnull()) & (df['"사진이 잘 나와요"'].isnull())].index

df.drop(index_3, inplace= True)

In [ ]:
# 브런치의 경우, 뷰가 좋아요, 음식이 맛있어요, 가성비가 좋아요, 화장실이 깨끗해요, 친절해요, 매장이 청결해요가 있는 겨웅가 있었다.
# pc방의 경우 좌석이 편해요가 있고, 음식점은 특별한 메뉴가
# 정말 카페만 해당하는 키워드가 전부 NaN인 경우는 전체 제거하는 것도 나쁘지 않아보인다.


# 515 티룸은 망했네
# 떡집도 제거?? => 키워드가 달라서 제거
# 써브웨이, 용호야채곱창, 로스니버거, 연소바,이삭토스트, 산밑에, 브런치가, 인비트원(샐러드), 카페마마스(양식).
# 일품칼국수, 로코민트
# 더올리브는 리뷰가 10개가 아닌데 갖고옴.. 와이파이 끊길때 정보 가져온듯
# 카페, 커피라고 적혀있으나 네이버지도에는 브런치로 분류된 경우 키워드 종류가 달라져서 이 경우 통일된 데이터셋을 얻기에는 방해가 되는 것 같아 전부 제거했다.

df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) &
   (df['"사진이 잘 나와요"'].isnull())][150:200]

name         search_name address  \
6961           뉴욕핫도그앤커피 잠실점      뉴욕핫도그앤커피 롯데월드점     잠실동   
6968             삼차 방배카페골목점          삼차 방배카페골목점     방배동   
6990                  카페395    밀레니엄 힐튼 서울 카페395  남대문로5가   
7002                오른손푸드카페             오른손푸드카페     봉천동   
7071                     잭팟            오스테리아 잭팟     공릉동   
7105             청년다방 롯데월드점          청년다방 롯데월드점     잠실동   
7179                     그램                  그램     양재동   
7210             청년다방 송파방이점          청년다방 송파방이점     방이동   
7256                Mr.goat               미스터고트     신원동   
7272          청년다방 구로지하이시티점       청년다방 구로지하이시티점     구로동   
7282                    주부리                 주부리     방배동   
7303                    누리정                 누리정     상암동   
7322               스테이지28카페              스테이지28     고덕동   
7369            청년다방 강남구청역점         청년다방 강남구청역점     논현동   
7444                  미스꼬레아               미스꼬레아     사당동   
7463           뉴욕핫도그앤커피 마포점     뉴욕핫도그앤커피 마포대흥역점     염리동   
7476   오리지널팬케이크하우스이태원점 이태원점    오리지널팬케이크하우스 이태원점    이태원동   
7522            날쌘카페 상암DMC점         날쌘카페 상암DMC점     증산동   
7532            청년다방가락점 가락점            청년다방 가락점     가락동   
7581               청년다방 강변점            청년다방 강변점     구의동   
7607                샐러디 방이점             샐러디 방이점     방이동   
7626        카페마마스잠실점 마마스잠실점           카페마마스 잠실점     신천동   
7679                   니들누들            니들누들 면목점      목동   
7729               더라운드 삼성점            더라운드 삼성점     삼성동   
7762                   Maon                Maon   보문동6가   
7769               닥터로빈 이대점            닥터로빈 이대점     대현동   
7782               닥터로빈 이대점            닥터로빈 이대점     대현동   
7790                    일오공                 150     논현동   
7804                   현대다방          청년다방 봉천현대점     용두동   
7853  카페마마스상암MBC점 마마스상암MBC점        카페마마스 상암MBC점     상암동   
7915                   커피.3             coffee3    노량진동   
7927                     릴즈                  릴즈   하왕십리동   
7932             BANANATALK               바나나토크     군자동   
7938           청년다방 문정법조타운점        청년다방 문정법조타운점     문정동   
7980              SALMOS231           SALMOS231     양재동   
7985                 비스트로33              비스트로33     신원동   
8055                   몰토베네                몰토베네     합정동   
8109                     모뜨                  모뜨     평창동   
8133                    무제한                 무제한     서교동   
8145                  베이스캠프               베이스캠프     우이동   
8147                    지엠유                 지엠유     한남동   
8162            청년다방 마곡나루역점         청년다방 마곡나루역점     마곡동   
8180      아메리칸에그샌드 롯데몰김포공항점  아메리칸에그샌드 롯데몰 김포공항점     방화동   
8244          에그드랍마곡역점 마곡역점           에그드랍 마곡역점     마곡동   
8281                   베어스덴            베어스덴 브런치     창천동   
8283                   파넬카페                  파넬     신사동   
8298               메종갤러리아한남           고메이494 한남     한남동   
8301             Folkcoffee                포크커피     구산동   
8318                   더라운지          The Lounge    노량진동   
8319              에그드랍 보라매점           에그드랍 보라매점    신대방동   

                            road_address  near_station station_distance  \
6961             서울 송파구 올림픽로 240 어드벤처 4층         28잠실역              52m   
6968                    서울 서초구 방배중앙로 151         47이수역             697m   
6990                서울 중구 소월로 50 메인로비 1층          4회현역             387m   
7002                서울 관악구 관악로14길 111 1층         2낙성대역             508m   
7071                                 NaN           NaN              NaN   
7105           서울 송파구 올림픽로 240 롯데월드 지하1층         28잠실역              52m   
7179                  서울 서초구 논현로31길 6 1층       3신분당양재역             749m   
7210                    서울 송파구 오금로31길 10          5방이역             581m   
7256                     서울 서초구 청룡마을길 13     신분당청계산입구역             448m   
7272                     서울 구로구 디지털로 243         7남구로역             800m   
7282            서울 서초구 동광로12가길 25 1층 주부리         47이수역             155m   
7303                    서울 마포구 월드컵북로 396       경의중앙수색역             694m   
7322                                 N

In [ ]:
index4 = df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull()) &
   (df['"사진이 잘 나와요"'].isnull())].index

df.drop(index=index4, inplace = True)

In [ ]:
# 데미타스는 잘못 불러와서 제거 

index5 = df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull()) & (df['"대화하기 좋아요"'].isnull()) & 
   (df['"집중하기 좋아요"'].isnull())].index

df.drop(index=index5, inplace=True)

### 음료, 디저트와 관련된 키워드를 안 갖고 있는 경우 제거

In [ ]:
# 음료, 디저트에 해당하는 부분만 전부 NaN인 경우를 보니까 bar도 많이 걸러지는듯
df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull())]

name        search_name address  \
115               더연           더연카페&Bar    종로2가   
180            종로abc              종로abc     관철동   
335        망우다방&일상맥주          망우다방&일상맥주     망우동   
362              오너스                오너스     역삼동   
424              커피락               커피락펍    서소문동   
629             클라시코               클라시코     신사동   
711               히읗                 히읗     쌍문동   
822           잭팟잉글리쉬         BLACK&JACK     서교동   
979           슈타인도르프             슈타인도르프     방이동   
1001           디바야누스              디바야누스     서초동   
1260   카페어퍼콜라보 어퍼콜라보              어퍼콜라보     한남동   
1263            엘파티오           El Patio   성수동1가   
1495            모아다방               모아다방   문래동2가   
1522               알             꿩먹고알먹고     서계동   
1843         산수화티하우스            산수화티하우스     한남동   
1881           루프탑지니              루프탑지니     동교동   
2211           천년동안도           천년동안도 낙원     낙원동   
2306          플로어포픽션  Floor For Fiction     소격동   
2404            안주다방               안주다방     구로동   
2429              몽아                 몽아     군자동   
2507             미나리                미나리     자양동   
2528         라이브클럽소우            라이브클럽소우     수송동   
2843            스프레드               스프레드    신대방동   
3040            운영카페               운영까페     상도동   
3144         스윗커피앤와인            스윗와인앤커피     구로동   
3752            남의커피                남의집     마곡동   
3851              시바             개편한 시바    이태원동   
3903             곰다방                곰다방     구로동   
4203          강남주막다방             강남주막다방     신사동   
4424             플랜힙                플랜힙     연남동   
4486           소프트컬러              소프트컬러     연남동   
4660              오우                 오우     주교동   
5335            펑키커피          고막힐링단펑키커피   성수동2가   
5460             테라스                테라스     양재동   
5480            스크립트               스크립트     창천동   
5641  36.5도여름동쪽점 동쪽점         36.5도여름동쪽점     서교동   
5796            커피락펍               커피락펍  남대문로4가   
5812             바우리                바우리     합정동   
6092   카페언플러그드 언플러그드              언플러그드     서교동   
6282            눈타이드               눈타이드    청량리동   
6381             산수화            산수화티하우스     한남동   
6651            와인웍스   와인웍스 현대백화점 압구정본점    압구정동   
6695             민석씨              민석씨커피     방배동   
6884   NorwegianWood             노르웨이의숲     수유동   
7132            비어위크               비어위크     봉천동   
7509         카페 더크림슨           카페 더 크림슨    이태원동   
7511           비닐하우스              비닐하우스   문래동3가   
7858         꾸까 광화문점      Terrace KUKKA     사간동   
8016              벨뷰                 벨뷰     청담동   
8457             톨릭스              카페톨릭스     도곡동   

                              road_address  near_station station_distance  \
115                   서울 종로구 종로 74 영안빌딩 5층          1종각역             174m   
180                서울 종로구 삼일대로17길 23 지하 1층          1종각역             302m   
335                  서울 중랑구 용마산로116길 14 1층     경의중앙경춘망우역             919m   
362                 서울 강남구 봉은사로4길 19 3층,4층      9신분당신논현역             190m   
424                      서울 중구 서소문로 134-14         12시청역             117m   
629                 서울 강남구 선릉로157길 27-3 B1   수인분당압구정로데오역             342m   
711                     서울 도봉구 우이천로 282 1층          4쌍문역             663m   
822                서울 마포구 와우산로29라길 11 1.5층  2공항경의중앙홍대입구역             260m   
979                       서울 송파구 오금로15길 11        9송파나루역             439m   
1001                서울 서초구 서초대로54길 9-8 길찾기         23교대역              83m   
1260                     서울 용산구 독서당로14길 29       경의중앙한남역             350m   
1263                      서울 성동구 연무장길 10-5          2뚝섬역             411m   
1495         서울 영등포구 도림로125가길 11-2 2층 201호          2문래역             455m   
1522                                   NaN           NaN              NaN   
1843                  서울 용산구 한남대로20길 21-14       경의중앙한남역             860m   
1881             서울 마포구 월드컵북로4길 23-5 3층,4층  2공항경의중앙홍대입구역             286m   
2211                     서울 종로구 수표로 134 2층      135종로3가역              96m   
2306               서울 종로구 북촌

In [ ]:
index6 = df[(df['"커피가 맛있어요"'].isnull()) & (df['"음료가 맛있어요"'].isnull()) &
   (df['"디저트가 맛있어요"'].isnull())].index

df.drop(index=index6, inplace=True)

In [ ]:
# 카페에 해당하는 경우만 남은 dataframe => 따로 저장해놓기
df

name   search_name address                road_address  \
0                     버블베어          버블베어     방화동  서울 강서구 금낭화로 167 방화8단지개화아파트   
1     커피빈코리아낙성대역점 코리아낙성대역점     커피빈 낙성대역점     봉천동      서울 관악구 남부순환로 1909 1,2층   
2                    노란고무줄    노란꼬무줄 선릉본점     대치동   서울 강남구 선릉로 424 1층5호 노란꼬무줄   
3                      커피랩           커피랩     서교동           서울 마포구 와우산로29길 14   
4            커핀그루나루신천점 신천점    커핀그루나루 신천점     잠실동            서울 송파구 올림픽로12길 8   
...                    ...           ...     ...                         ...   
8491            커피나무 보라매역점    커피나무 보라매역점     신길동           서울 영등포구 여의대방로 135   
8492            스노잉합정점 합정점       스노잉 합정점     합정동               서울 마포구 성지길 22   
8493          이디야커피 논골사거리점  이디야커피 논골사거리점   금호동1가               서울 성동구 난계로 35   
8494                 퍼블릭맨션         퍼블릭맨션      목동              서울 양천구 등촌로 154   
8495          이디야커피 굽은다리역점  이디야커피 굽은다리역점     천호동       서울 강동구 양재대로 1543 진주빌딩   

      near_station station_distance  total_review  "커피가 맛있어요"  "친절해요"  \
0             5방화역             427m            18         7.0     5.0   
1            2낙성대역             119m           284         7.0    56.0   
2         2수인분당선릉역             157m           134        40.0    14.0   
3     2공항경의중앙홍대입구역             324m           100        76.0    23.0   
4           2잠실새내역             228m           140        64.0    14.0   
...            ...              ...           ...         ...     ...   
8491       7신림보라매역              44m           219         0.0    40.0   
8492         26합정역             140m            48        21.0    17.0   
8493          5행당역             395m           189         1.0    46.0   
8494          9등촌역             825m            27        17.0    13.0   
8495        5굽은다리역             227m            65        28.0    20.0   

      "음료가 맛있어요"  ...  "집중하기 좋아요"  "인테리어가 멋져요"  "좌석이 편해요"  "화장실이 깨끗해요"  \
0            5.0  ...         NaN          NaN        1.0          2.0   
1           74.0  ...        48.0          8.0       33.0          2.0   
2           61.0  ...         NaN          1.0        NaN          NaN   
3           13.0  ...        10.0         19.0        2.0          6.0   
4           31.0  ...         7.0          2.0       15.0          5.0   
...          ...  ...         ...          ...        ...          ...   
8491        38.0  ...         8.0         16.0       28.0          8.0   
8492        11.0  ...         1.0          6.0        3.0          NaN   
8493        38.0  ...        21.0          5.0       21.0          3.0   
8494         9.0  ...         2.0          8.0        1.0          1.0   
8495        12.0  ...         3.0          3.0        NaN          NaN   

      "특별한 메뉴가 있어요"  "가성비가 좋아요"  "뷰가 좋아요"  "주차하기 편해요"  "사진이 잘 나와요"   same  
0               2.0         NaN       NaN         1.0          NaN   True  
1               2.0         8.0      14.0         NaN          1.0  False  
2              21.0        33.0       NaN         1.0          NaN  False  
3              10.0         4.0       5.0         NaN          4.0   True  
4               5.0         6.0       3.0         NaN          1.0  False  
...             ...         ...       ...         ...          ...    ...  
8491            7.0        23.0       7.0         3.0          1.0   True  
8492            5.0         3.0       NaN         NaN          2.0  False  
8493            5.0        28.0       2.0         2.0          6.0   True  
8494            2.0         1.0       NaN         NaN          NaN   True  
8495            NaN         3.0       1.0         1.0          1.0   True  

[8091 rows x 23 columns]

## 데이터 형식 전처리

In [ ]:
# 뒤에거 돌리다가 추가된 열이어서 제거하기
del df['same']

In [ ]:
# 리뷰데이터의 nan값을 0으로 바꾸기
df.loc[:,'"커피가 맛있어요"':] = df.loc[:,'"커피가 맛있어요"':].fillna(0)

### 'station_distance' 행 관련

In [ ]:
# station_distance을 사용하기 위해선 string이 아니라 integer이어야 함
# unique를 통해 속한 값 찾기

df['station_distance'].unique()

array(['427m', '119m', '157m', '324m', '228m', '200m', '156m', '333m',
       '203m', nan, '650m', '232m', '95m', '220m', '496m', '395m', '56m',
       '547m', '153m', '186m', '405m', '159m', '198m', '498m', '466m',
       '535m', '807m', '100m', '183m', '101m', '77m', '79m', '148m',
       '181m', '29m', '14m', '188m', '599m', '76m', '99m', '425m', '889m',
       '330m', '475m', '941m', '118m', '25m', '292m', '307m', '369m',
       '154m', '221m', '141m', '134m', '495m', '389m', '177m', '702m',
       '560m', '293m', '84m', '317m', '468m', '437m', '373m', '403m',
       '471m', '500m', '805m', '541m', '513m', '456m', '745m', '552m',
       '326m', '111m', '578m', '555m', '15m', '300m', '358m', '297m',
       '227m', '75m', '372m', '434m', '518m', '421m', '767m', '312m',
       '574m', '291m', '401m', '256m', '308m', '342m', '192m', '829m',
       '187m', '140m', '23m', '458m', '444m', '710m', '132m', '309m',
       '597m', '661m', '295m', '666m', '150m', '135m', '911m', '35m',
       

In [ ]:
# unique 부분을 보니 역이라는 걸 잘못 불러온 게 존재함 => 전처리로 제거하거나 직접 입력하기
df[df['station_distance']=='역']

name      search_name address  \
717     스타벅스 광화문우체국점     스타벅스 광화문우체국점     서린동   
2357           모닝그루브            모닝그루브     성내동   
2681  스타벅스 선유도역1번출구점  스타벅스 선유도역 1번출구점   양평동5가   
3104           공간다반사            공간다반사   당산동6가   

                           road_address near_station station_distance  \
717                         서울 종로구 종로 6        12시청역                역   
2357  서울 강동구 천호대로 1116 라이저 강동빌딩 1층 103호        5둔촌동역                역   
2681                    서울 영등포구 양평로 128        9선유도역                역   
3104                서울 영등포구 선유로55길 3 1층        29당산역                역   

      total_review  "커피가 맛있어요"  "친절해요"  "음료가 맛있어요"  ...  "대화하기 좋아요"  \
717           1561         7.0     3.0         1.0  ...         4.0   
2357           131        60.0    36.0        24.0  ...        20.0   
2681          1028         4.0     9.0         7.0  ...        94.0   
3104           196        86.0    58.0        59.0  ...        30.0   

      "집중하기 좋아요"  "인테리어가 멋져요"  "좌석이 편해요"  "화장실이 깨끗해요"  "특별한 메뉴가 있어요"  \
717         96.0         88.0       78.0         50.0           51.0   
2357        12.0         16.0       19.0          4.0           11.0   
2681        85.0         65.0       70.0         33.0           24.0   
3104        16.0         79.0        9.0          1.0           10.0   

      "가성비가 좋아요"  "뷰가 좋아요"  "주차하기 편해요"  "사진이 잘 나와요"  
717         60.0      38.0         2.0         22.0  
2357         7.0       1.0         4.0          2.0  
2681        24.0      24.0         2.0         16.0  
3104        14.0      26.0         0.0         32.0  

[4 rows x 22 columns]

In [ ]:
# unique 부분을 보니 역이라는 걸 잘못 불러온 게 존재함 => 전처리로 제거하거나 직접 입력하기
df[df['station_distance']=='역'].index

Int64Index([717, 2357, 2681, 3104], dtype='int64')

In [ ]:
# 직접 검색해서 값 채우기
df['station_distance'][717] = '78m'
df['station_distance'][2357] = '238m'
df['station_distance'][2681] = '5m'
df['station_distance'][3104] = '510m'

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from 

In [ ]:
# 제대로 수정되었는지 확인
df['station_distance'].unique()

array(['427m', '119m', '157m', '324m', '228m', '200m', '156m', '333m',
       '203m', nan, '650m', '232m', '95m', '220m', '496m', '395m', '56m',
       '547m', '153m', '186m', '405m', '159m', '198m', '498m', '466m',
       '535m', '807m', '100m', '183m', '101m', '77m', '79m', '148m',
       '181m', '29m', '14m', '188m', '599m', '76m', '99m', '425m', '889m',
       '330m', '475m', '941m', '118m', '25m', '292m', '307m', '369m',
       '154m', '221m', '141m', '134m', '495m', '389m', '177m', '702m',
       '560m', '293m', '84m', '317m', '468m', '437m', '373m', '403m',
       '471m', '500m', '805m', '541m', '513m', '456m', '745m', '552m',
       '326m', '111m', '578m', '555m', '15m', '300m', '358m', '297m',
       '227m', '75m', '372m', '434m', '518m', '421m', '767m', '312m',
       '574m', '291m', '401m', '256m', '308m', '342m', '192m', '829m',
       '187m', '140m', '23m', '458m', '444m', '710m', '132m', '309m',
       '597m', '661m', '295m', '666m', '150m', '135m', '911m', '35m',
       

In [ ]:
# 전부 m 단위여서 m을 제거해 변수명에 추가하고, integer로 바꾸기
df["station_distance"] = df["station_distance"].str.replace(pat='m', repl=r'', regex=True)

In [ ]:
# 제대로 수정되었는지 확인
df['station_distance'].unique()

array(['427', '119', '157', '324', '228', '200', '156', '333', '203', nan,
       '650', '232', '95', '220', '496', '395', '56', '547', '153', '186',
       '405', '159', '198', '498', '466', '535', '807', '100', '183',
       '101', '77', '79', '148', '181', '29', '14', '188', '599', '76',
       '99', '425', '889', '330', '475', '941', '118', '25', '292', '307',
       '369', '154', '221', '141', '134', '495', '389', '177', '702',
       '560', '293', '84', '317', '468', '437', '373', '403', '471',
       '500', '805', '541', '513', '456', '745', '552', '326', '111',
       '578', '555', '15', '300', '358', '297', '227', '75', '372', '434',
       '518', '421', '767', '312', '574', '291', '401', '256', '308',
       '342', '192', '829', '187', '140', '23', '458', '444', '710',
       '132', '309', '597', '661', '295', '666', '150', '135', '911',
       '35', '49', '364', '19', '750', '930', '660', '396', '510', '748',
       '448', '391', '708', '10', '430', '120', '104', '418', '97'

In [ ]:
# 저장하기
df.to_csv('naver_cafe_all.csv',index=False)

### 의미없는 행 제거

In [ ]:
# 불러오기
df = pd.read_csv('naver_cafe_all.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8091 entries, 0 to 8090
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              8091 non-null   object 
 1   search_name       8091 non-null   object 
 2   address           8091 non-null   object 
 3   road_address      7258 non-null   object 
 4   near_station      7258 non-null   object 
 5   station_distance  7258 non-null   float64
 6   total_review      8091 non-null   int64  
 7   "커피가 맛있어요"        8091 non-null   float64
 8   "친절해요"            8091 non-null   float64
 9   "음료가 맛있어요"        8091 non-null   float64
 10  "디저트가 맛있어요"       8091 non-null   float64
 11  "매장이 청결해요"        8091 non-null   float64
 12  "대화하기 좋아요"        8091 non-null   float64
 13  "집중하기 좋아요"        8091 non-null   float64
 14  "인테리어가 멋져요"       8091 non-null   float64
 15  "좌석이 편해요"         8091 non-null   float64
 16  "화장실이 깨끗해요"       8091 non-null   float64


In [ ]:
# 승주가 부탁한 작업 진행
df['cafe'] = df['name'] + ' '+ df['address']

In [ ]:
df['cafe'].head()

0                    버블베어 방화동
1    커피빈코리아낙성대역점 코리아낙성대역점 봉천동
2                   노란고무줄 대치동
3                     커피랩 서교동
4           커핀그루나루신천점 신천점 잠실동
Name: cafe, dtype: object

In [ ]:
last_ref  = pd.read_csv('/content/drive/MyDrive/modeling/publicdata_cafe_lastdrop.csv')

In [ ]:
# 아 왜 안된다 생각했는데 역시 띄어쓰기가 되어있었네...
last_ref['cafe'].head()

0    커피빈 코리아대학로대명거리점 명륜4가
1           요거프레소 쌍문점 쌍문동
2               우성커피숍 신월동
3                버블베어 방화동
4                 알뤼르 대치동
Name: cafe, dtype: object

In [ ]:
df['address'].unique()

array(['방화동', '봉천동', '대치동', '서교동', '잠실동', '가산동', '서초동', '도곡동', '신길동',
       '여의도동', '청담동', '역삼동', '동숭동', '신내동', '성산동', '구기동', '목동', '창천동',
       '사당동', '신림동', '양재동', '망원동', '진관동', '석관동', '행당동', '화곡동', '미아동',
       '부암동', '성수동2가', '당산동6가', '군자동', '평창동', '동자동', '신설동', '신사동',
       '광희동2가', '종암동', '개봉동', '내발산동', '월계동', '안암동5가', '문래동3가', '신수동',
       '잠원동', '삼성동', '대방동', '충정로3가', '무악동', '신문로2가', '반포동', '일원동',
       '한강로2가', '방배동', '가락동', '천호동', '양평동4가', '논현동', '연희동', '누하동', '자양동',
       '등촌동', '장충동2가', '화양동', '정릉동', '도화동', '구로동', '수유동', '중학동', '문정동',
       '이촌동', '방이동', '관수동', '흑석동', '염리동', '개포동', '동빙고동', '냉천동', '동교동',
       '안국동', '갈현동', '연남동', '성수동1가', '남창동', '상계동', '후암동', '홍제동', '시흥동',
       '소격동', '중곡동', '불광동', '신원동', '창동', '고척동', '중림동', '공릉동', '을지로2가',
       '길음동', '원효로1가', '합정동', '명륜1가', '만리동1가', '한남동', '순화동', '명동2가',
       '제기동', '신천동', '송파동', '문래동2가', '마곡동', '성내동', '신당동', '영등포동7가',
       '북가좌동', '답십리동', '용두동', '당산동4가', '장안동', '염창동', '길동', '미근동', '을지로6가',
       '충정로2가'

In [ ]:
df.head()

name search_name address                road_address  \
0                  버블베어        버블베어     방화동  서울 강서구 금낭화로 167 방화8단지개화아파트   
1  커피빈코리아낙성대역점 코리아낙성대역점   커피빈 낙성대역점     봉천동      서울 관악구 남부순환로 1909 1,2층   
2                 노란고무줄  노란꼬무줄 선릉본점     대치동   서울 강남구 선릉로 424 1층5호 노란꼬무줄   
3                   커피랩         커피랩     서교동           서울 마포구 와우산로29길 14   
4         커핀그루나루신천점 신천점  커핀그루나루 신천점     잠실동            서울 송파구 올림픽로12길 8   

   near_station  station_distance  total_review  "커피가 맛있어요"  "친절해요"  \
0          5방화역             427.0            18         7.0     5.0   
1         2낙성대역             119.0           284         7.0    56.0   
2      2수인분당선릉역             157.0           134        40.0    14.0   
3  2공항경의중앙홍대입구역             324.0           100        76.0    23.0   
4        2잠실새내역             228.0           140        64.0    14.0   

   "음료가 맛있어요"  ...  "집중하기 좋아요"  "인테리어가 멋져요"  "좌석이 편해요"  "화장실이 깨끗해요"  \
0         5.0  ...         0.0          0.0        1.0          2.0   
1        74.0  ...        48.0          8.0       33.0          2.0   
2        61.0  ...         0.0          1.0        0.0          0.0   
3        13.0  ...        10.0         19.0        2.0          6.0   
4        31.0  ...         7.0          2.0       15.0          5.0   

   "특별한 메뉴가 있어요"  "가성비가 좋아요"  "뷰가 좋아요"  "주차하기 편해요"  "사진이 잘 나와요"  \
0            2.0         0.0       0.0         1.0          0.0   
1            2.0         8.0      14.0         0.0          1.0   
2           21.0        33.0       0.0         1.0          0.0   
3           10.0         4.0       5.0         0.0          4.0   
4            5.0         6.0       3.0         0.0          1.0   

                       cafe  
0                  버블베어 방화동  
1  커피빈코리아낙성대역점 코리아낙성대역점 봉천동  
2                 노란고무줄 대치동  
3                   커피랩 서교동  
4         커핀그루나루신천점 신천점 잠실동  

[5 rows x 23 columns]

In [ ]:
last_ref.head()

상가업소번호    상호명          지점명 상권업종대분류코드 상권업종대분류명 상권업종중분류코드 상권업종중분류명  \
0  20659878    커피빈  코리아대학로대명거리점         Q       음식       Q12   커피점/카페   
1  20676428  요거프레소          쌍문점         Q       음식       Q12   커피점/카페   
2  28491095  우성커피숍          NaN         Q       음식       Q12   커피점/카페   
3  23260840   버블베어          NaN         Q       음식       Q12   커피점/카페   
4  22206830    알뤼르          NaN         Q       음식       Q12   커피점/카페   

  상권업종소분류코드     상권업종소분류명 표준산업분류코드  ...                도로명주소   구우편번호   신우편번호  \
0    Q12A01  커피전문점/카페/다방   I56220  ...     서울특별시 종로구 대명길 25  110524  3078.0   
1    Q12A01  커피전문점/카페/다방   I56220  ...      서울특별시 도봉구 도당로 7  132030  1385.0   
2    Q12A01  커피전문점/카페/다방   I56220  ...     서울특별시 양천구 오목로 48  158090  8029.0   
3    Q12A01  커피전문점/카페/다방   I56220  ...   서울특별시 강서구 금낭화로 167  157788  7509.0   
4    Q12A01  커피전문점/카페/다방   I56220  ...  서울특별시 강남구 삼성로85길 33  135280  6196.0   

   동정보  층정보  호정보          경도         위도               이름                  cafe  
0  NaN    3  NaN  127.000171  37.583149  커피빈 코리아대학로대명거리점  커피빈 코리아대학로대명거리점 명륜4가  
1  NaN    1  NaN  127.034746  37.658728        요거프레소 쌍문점         요거프레소 쌍문점 쌍문동  
2  NaN    1  NaN  126.842935  37.522055            우성커피숍             우성커피숍 신월동  
3  NaN    1  NaN  126.813358  37.580941             버블베어              버블베어 방화동  
4  NaN  NaN  NaN  127.054731  37.504378              알뤼르               알뤼르 대치동  

[5 rows x 41 columns]

In [ ]:
# 이제 종류는 카페인데 그 카페를 잘못 가져온 경우만 전처리 진행하면 됨. # 근데 오히려 행의 수가 늘어났다??? 중복인 친구가 아직 있는듯..?
pd.merge(right = last_ref, left = df, on='cafe')['cafe']

name   search_name address                road_address  \
0                     버블베어          버블베어     방화동  서울 강서구 금낭화로 167 방화8단지개화아파트   
1     커피빈코리아낙성대역점 코리아낙성대역점     커피빈 낙성대역점     봉천동      서울 관악구 남부순환로 1909 1,2층   
2                    노란고무줄    노란꼬무줄 선릉본점     대치동   서울 강남구 선릉로 424 1층5호 노란꼬무줄   
3                      커피랩           커피랩     서교동           서울 마포구 와우산로29길 14   
4            커핀그루나루신천점 신천점    커핀그루나루 신천점     잠실동            서울 송파구 올림픽로12길 8   
...                    ...           ...     ...                         ...   
8093            커피나무 보라매역점    커피나무 보라매역점     신길동           서울 영등포구 여의대방로 135   
8094            스노잉합정점 합정점       스노잉 합정점     합정동               서울 마포구 성지길 22   
8095          이디야커피 논골사거리점  이디야커피 논골사거리점   금호동1가               서울 성동구 난계로 35   
8096                 퍼블릭맨션         퍼블릭맨션      목동              서울 양천구 등촌로 154   
8097          이디야커피 굽은다리역점  이디야커피 굽은다리역점     천호동       서울 강동구 양재대로 1543 진주빌딩   

      near_station  station_distance  total_review  "커피가 맛있어요"  "친절해요"  \
0             5방화역             427.0            18         7.0     5.0   
1            2낙성대역             119.0           284         7.0    56.0   
2         2수인분당선릉역             157.0           134        40.0    14.0   
3     2공항경의중앙홍대입구역             324.0           100        76.0    23.0   
4           2잠실새내역             228.0           140        64.0    14.0   
...            ...               ...           ...         ...     ...   
8093       7신림보라매역              44.0           219         0.0    40.0   
8094         26합정역             140.0            48        21.0    17.0   
8095          5행당역             395.0           189         1.0    46.0   
8096          9등촌역             825.0            27        17.0    13.0   
8097        5굽은다리역             227.0            65        28.0    20.0   

      "음료가 맛있어요"  ...         건물명                 도로명주소   구우편번호   신우편번호  동정보  \
0            5.0  ...  방화8단지개화아파트    서울특별시 강서구 금낭화로 167  157788  7509.0  NaN   
1           74.0  ...         NaN  서울특별시 관악구 남부순환로 1909  151050  8742.0  NaN   
2           61.0  ...         NaN     서울특별시 강남구 선릉로 424  135280  6198.0  NaN   
3           13.0  ...         NaN  서울특별시 마포구 와우산로29길 14  121836  4052.0  NaN   
4           31.0  ...        제일빌딩   서울특별시 송파구 올림픽로12길 8  138220  5557.0  NaN   
...          ...  ...         ...                   ...     ...     ...  ...   
8093        38.0  ...         NaN  서울특별시 영등포구 여의대방로 135  150842  7432.0  NaN   
8094        11.0  ...         NaN      서울특별시 마포구 성지길 22  121897  4071.0  NaN   
8095        38.0  ...         NaN      서울특별시 성동구 난계로 35  133800  4727.0  NaN   
8096         9.0  ...         NaN     서울특별시 양천구 등촌로 154  158050  7956.0  NaN   
8097        12.0  ...        진주빌딩   서울특별시 강동구 양재대로 1543  134020  5315.0  NaN   

      층정보  호정보          경도         위도                    이름  
0       1  NaN  126.813358  37.580941                  버블베어  
1       1  NaN  126.961593  37.478106  커피빈코리아낙성대역점 코리아낙성대역점  
2       1  NaN  127.050045  37.502899                 노란고무줄  
3       1  NaN  126.929223  37.554964                   커피랩  
4       1  NaN  127.082767  37.510988         커핀그루나루신천점 신천점  
...   ...  ...         ...        ...                   ...  
8093    1  NaN  126.920045  37.499187            커피나무 보라매역점  
8094    1  NaN  126.913098  37.547717            스노잉합정점 합정점  
8095    1  NaN  127.024466  37.555519          이디야커피 논골사거리점  
8096    1  NaN  126.862576  37.543529                 퍼블릭맨션  
8097    1  NaN  127.141847  37.543501          이디야커피 굽은다리역점  

[8098 rows x 63 columns]

In [ ]:
## 다른 코드로 진행해보자
del_cafe = pd.read_csv('/content/drive/MyDrive/modeling/publicdata_deleted_cafe.csv')

for i in range(len(del_cafe)):
  df.drop(index = df[df['cafe']==del_cafe['cafe'][i]].index, inplace=True)
  print(len(df))

8090
8090
8090
8090
8089
8089
8088
8087
8086
8086
8086
8085
8085
8085
8085
8084
8084
8084
8083
8083
8082
8082
8081
8080
8079
8078
8077
8077
8077
8076
8076
8076
8076
8076
8076
8076
8076
8076
8076
8076
8076
8075
8075
8075
8074
8074
8073
8073
8072
8072
8072
8071
8071
8071
8071
8071
8071
8071
8071
8071
8071
8071
8070
8069
8068
8067
8067
8066
8066
8065
8064
8063
8062
8061
8061
8060
8060
8059
8059
8059
8058
8057
8057
8057
8057
8057
8057
8057
8057
8056
8056
8055
8055
8055
8054
8053
8053
8053
8053
8052
8051
8050
8049
8048
8047
8046
8046
8045
8044
8044
8043
8043
8042
8042
8042
8042
8042
8042
8042
8041
8041
8041
8040
8039
8038
8037
8036
8036
8035
8034
8034
8033
8033
8032
8031
8030
8029
8028
8027
8027
8026
8025
8024
8023
8022
8022
8022
8022
8021
8021
8020
8020
8019
8019
8019
8018
8017
8016
8015
8014
8014
8014
8013
8013
8013
8013
8012
8011
8010
8010
8010
8009
8008
8008
8007
8006


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8006 entries, 0 to 8090
Data columns (total 23 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              8006 non-null   object 
 1   search_name       8006 non-null   object 
 2   address           8006 non-null   object 
 3   road_address      7179 non-null   object 
 4   near_station      7179 non-null   object 
 5   station_distance  7179 non-null   float64
 6   total_review      8006 non-null   int64  
 7   "커피가 맛있어요"        8006 non-null   float64
 8   "친절해요"            8006 non-null   float64
 9   "음료가 맛있어요"        8006 non-null   float64
 10  "디저트가 맛있어요"       8006 non-null   float64
 11  "매장이 청결해요"        8006 non-null   float64
 12  "대화하기 좋아요"        8006 non-null   float64
 13  "집중하기 좋아요"        8006 non-null   float64
 14  "인테리어가 멋져요"       8006 non-null   float64
 15  "좌석이 편해요"         8006 non-null   float64
 16  "화장실이 깨끗해요"       8006 non-null   float64


## 네이버지도와 원래 데이터셋의 주소 비교 (실제 장소도 같은 카페인가)

In [ ]:
# csv 파일 저장

# df.to_csv('naver_processed.csv', index= False)

In [ ]:
df = pd.read_csv('naver_preprocessed.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7960 entries, 0 to 7959
Data columns (total 24 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        7960 non-null   int64  
 1   name              7960 non-null   object 
 2   search_name       7960 non-null   object 
 3   address           7960 non-null   object 
 4   road_address      7142 non-null   object 
 5   near_station      7142 non-null   object 
 6   station_distance  7142 non-null   float64
 7   total_review      7960 non-null   int64  
 8   "커피가 맛있어요"        7960 non-null   float64
 9   "친절해요"            7960 non-null   float64
 10  "음료가 맛있어요"        7960 non-null   float64
 11  "디저트가 맛있어요"       7960 non-null   float64
 12  "매장이 청결해요"        7960 non-null   float64
 13  "대화하기 좋아요"        7960 non-null   float64
 14  "집중하기 좋아요"        7960 non-null   float64
 15  "인테리어가 멋져요"       7960 non-null   float64
 16  "좌석이 편해요"         7960 non-null   float64


In [ ]:
df['cafe'].nunique()

7960

In [ ]:
df['cafe'].value_counts()[:45]

8006개 중 7960개의 'cafe' 값들이 있으니 value_counts로 46개를 봐보자.

In [ ]:
# 중복되는 이름을 가지고 for 문을 돌려볼까?
idx_cafe_nonuniq = df['cafe'].value_counts()[:45].index
print(df[df['cafe'] == idx_cafe_nonuniq[0]])

In [ ]:
df.loc[:,['name','search_name']]

name   search_name
0                     버블베어          버블베어
1     커피빈코리아낙성대역점 코리아낙성대역점     커피빈 낙성대역점
2                    노란고무줄    노란꼬무줄 선릉본점
3                      커피랩           커피랩
4            커핀그루나루신천점 신천점    커핀그루나루 신천점
...                    ...           ...
8001            커피나무 보라매역점    커피나무 보라매역점
8002            스노잉합정점 합정점       스노잉 합정점
8003          이디야커피 논골사거리점  이디야커피 논골사거리점
8004                 퍼블릭맨션         퍼블릭맨션
8005          이디야커피 굽은다리역점  이디야커피 굽은다리역점

[8006 rows x 2 columns]

In [ ]:
# 이 경우 전부 동일한 카페가 뽑힌 것을 알 수 있음 => 전부 하나만 남기기

for i in range(45):
    print(df.loc[:,['cafe','search_name','road_address']][df['cafe'] == idx_cafe_nonuniq[i]])

          cafe search_name           road_address
1040  커피바이 잠실동   커피바이 로스터스  서울 송파구 백제고분로17길 56 2층
6737  커피바이 잠실동   커피바이 로스터스  서울 송파구 백제고분로17길 56 2층
7311  커피바이 잠실동   커피바이 로스터스  서울 송파구 백제고분로17길 56 2층
                    cafe   search_name    road_address
5614  스타벅스 고대안암병원점 안암동5가  스타벅스 고대안암병원점  서울 성북구 고려대로 73
7882  스타벅스 고대안암병원점 안암동5가  스타벅스 고대안암병원점  서울 성북구 고려대로 73
            cafe search_name            road_address
167   카페조이 성수동2가        카페조이  서울 성동구 아차산로11길 11 105호
1546  카페조이 성수동2가        카페조이  서울 성동구 아차산로11길 11 105호
         cafe search_name       road_address
406  심플커피 장위동   심플커피 장위본점  서울 성북구 돌곶이로25길 85
789  심플커피 장위동   심플커피 장위본점  서울 성북구 돌곶이로25길 85
             cafe search_name road_address
35    드롭오가닉커피 부암동   드롭 오가닉 커피          NaN
5352  드롭오가닉커피 부암동   드롭 오가닉 커피          NaN
         cafe search_name road_address
5384  노아스 연희동  노아스로스팅 연희점          NaN
5534  노아스 연희동  노아스로스팅 연희점          NaN
           cafe search_name              road_address
1849  카페루루 노량진동        카페루루  서울 동작구 등용로14

In [ ]:
# 46 개 추가로 제거한다.
df = df.drop_duplicates(['cafe'])
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7960 entries, 0 to 8005
Data columns (total 23 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   name              7960 non-null   object 
 1   search_name       7960 non-null   object 
 2   address           7960 non-null   object 
 3   road_address      7142 non-null   object 
 4   near_station      7142 non-null   object 
 5   station_distance  7142 non-null   float64
 6   total_review      7960 non-null   int64  
 7   "커피가 맛있어요"        7960 non-null   float64
 8   "친절해요"            7960 non-null   float64
 9   "음료가 맛있어요"        7960 non-null   float64
 10  "디저트가 맛있어요"       7960 non-null   float64
 11  "매장이 청결해요"        7960 non-null   float64
 12  "대화하기 좋아요"        7960 non-null   float64
 13  "집중하기 좋아요"        7960 non-null   float64
 14  "인테리어가 멋져요"       7960 non-null   float64
 15  "좌석이 편해요"         7960 non-null   float64
 16  "화장실이 깨끗해요"       7960 non-null   float64


In [ ]:
df['cafe'].value_counts()[:45] # 이젠 다 제거되었음

버블베어 방화동                   1
에이쓰리커피집 신사동                1
공차구로디지털2호점 구로디지털2호점 구로동    1
REPUBLICOFCOFFEE 마곡동       1
스타벅스포스코사거리점 포스코사거리점 삼성동    1
밀랑스 대흥동                    1
페라 대현동                     1
카페빅컵 성산동                   1
공차 약수점 신당동                 1
스타벅스 대치대원빌딩R점 대치동          1
커피스트 신문로2가                 1
투썸플레이스충정로역점 충정로역점 충정로3가    1
커피정원 상계동                   1
아마스빈 버블티우장산역점 내발산동         1
스타일커피 삼성동                  1
튠업카페 이문동                   1
이디야커피 청구역점 신당동             1
더스키&커피 동숭동                 1
커피폴 구산동                    1
가나슈블랫 연남동                  1
2막 창성동                     1
이디야커피 용두동사거리점 용두동          1
크레센도 봉천동                   1
만돌다방 길동                    1
CREMADORO 우면동              1
카페오로시 자양동                  1
카페마리 북아현동                  1
이디야커피 신촌역점 창천동             1
더베스트커피 가양동                 1
작업실 소소한 망원동                1
카페902 돈암동                  1
이디야커피 오장동점 오장동             1
투썸플레이스연신내점 연신내점 갈현동        1
셀렉토커피동묘역점 동묘역점 숭인동         1
후아나 월계동       

## 소상공인 데이터셋의 카페명을 제거한 데이터셋

In [ ]:
## 데이터 분석 목적에 맞춰 crawling 한 결과만을 가지고 데이터셋을 만들어보았다. (추가 병합 과정 없을 예정하에)

# search_name과 'road_address'를 기준을 제거하면, 같은 결과로 크롤링된 데이터를 모두 제거할 수 있다.

df_s = df.drop_duplicates(['search_name', 'road_address'])
df_s.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7661 entries, 0 to 7959
Data columns (total 24 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        7661 non-null   int64  
 1   name              7661 non-null   object 
 2   search_name       7661 non-null   object 
 3   address           7661 non-null   object 
 4   road_address      6877 non-null   object 
 5   near_station      6877 non-null   object 
 6   station_distance  6877 non-null   float64
 7   total_review      7661 non-null   int64  
 8   "커피가 맛있어요"        7661 non-null   float64
 9   "친절해요"            7661 non-null   float64
 10  "음료가 맛있어요"        7661 non-null   float64
 11  "디저트가 맛있어요"       7661 non-null   float64
 12  "매장이 청결해요"        7661 non-null   float64
 13  "대화하기 좋아요"        7661 non-null   float64
 14  "집중하기 좋아요"        7661 non-null   float64
 15  "인테리어가 멋져요"       7661 non-null   float64
 16  "좌석이 편해요"         7661 non-null   float64


In [ ]:
# df_s 의 경우에는 크롤링된 정보만을 사용할 것이기에 name이나 그런 행들을 전부 제거하려 한다.

df_s = df_s.drop(['Unnamed: 0', 'name', 'cafe', ], axis=1)

In [ ]:
# 깔끔하게 행이 제거된 것을 확인할 수 있다.
df_s.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7661 entries, 0 to 7959
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   search_name       7661 non-null   object 
 1   address           7661 non-null   object 
 2   road_address      6877 non-null   object 
 3   near_station      6877 non-null   object 
 4   station_distance  6877 non-null   float64
 5   total_review      7661 non-null   int64  
 6   "커피가 맛있어요"        7661 non-null   float64
 7   "친절해요"            7661 non-null   float64
 8   "음료가 맛있어요"        7661 non-null   float64
 9   "디저트가 맛있어요"       7661 non-null   float64
 10  "매장이 청결해요"        7661 non-null   float64
 11  "대화하기 좋아요"        7661 non-null   float64
 12  "집중하기 좋아요"        7661 non-null   float64
 13  "인테리어가 멋져요"       7661 non-null   float64
 14  "좌석이 편해요"         7661 non-null   float64
 15  "화장실이 깨끗해요"       7661 non-null   float64
 16  "특별한 메뉴가 있어요"     7661 non-null   float64
